WebApp/src/main/java/ev/projects/webapp/services/DocumentService.java

```diff
@@ -0,0 +1,45 @@
+package ev.projects.webapp.services;
+
+import ev.projects.Document;
+import ev.projects.webapp.repositories.IDocumentRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.io.InputStream;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+
+@Service
+public class DocumentService implements IDocumentService {
+
+    private IDocumentRepository documentRepository;
+
+    @Autowired
+    public DocumentService(IDocumentRepository documentRepository) {
+        this.documentRepository = documentRepository;
+    }
+
+
+    @Override
+    public List<Document> getAllDocumentsByCase(long caseID) {
+        return documentRepository.getAllDocumentsByCase(caseID);
+    }
+
+    @Override
+    public List<Document> getAllAttachmentsByDocument(long documentID) {
+        return documentRepository.getAllAttachmentsByDocument(documentID);
+    }
+
+    @Override
+    public byte[] downloadDocument(long ID) throws Exception {
+        Optional<Document> document = documentRepository.getById(ID);
+        if(document.isPresent()) {
```

Comment:
I have prodived a pseudo code, sorry for puzzling you.
However the idea behind it was to use Optional instead of "if" conditionals.
It would require you to create some private methods and handle IOException inside.
The normal practise would be handling such exceptions and throwing RuntimeException instead.
The handled exceptions should not be thrown unless they get caught somewere later.

Reply:
Just noticed that the method has unused "throws Exception". Removed it.